custom_components/octopus_energy/discovery.py

```diff
@@ -66,7 +66,7 @@ async def _async_start_discovery(self) -> None:
         for item in entities:
             unique_id: str = item[1].unique_id
 
-            if "octopus_energy" in unique_id:
+            if unique_id is not None and "octopus_energy" in str(unique_id):
```

Comment:
It would probably be best to do this conversion on line 67 and update the typings incase something else needs it in the future.